src/frontend/src/meta_client.rs

```diff
@@ -134,6 +135,8 @@ pub trait FrontendMetaClient: Send + Sync {
     async fn list_rate_limits(&self) -> Result<Vec<RateLimitInfo>>;
 
     async fn get_meta_store_endpoint(&self) -> Result<String>;
+
+    async fn list_iceberg_tables(&self) -> Result<Vec<IcebergTable>>;
```

Comment:
nitpicking:
```suggestion
    async fn list_hosted_iceberg_tables(&self) -> Result<Vec<IcebergTable>>;
```
might be more clear

In meta part it's put in `HostedIcebergCatalogService`, but seems in meta_client here, we don't have that context